specification/keyvault/data-plane/Security.KeyVault.SecurityDomain/models.tsp

```diff
@@ -0,0 +1,226 @@
+import "@azure-tools/typespec-azure-core";
+import "@typespec/rest";
+import "@typespec/http";
+
+using Azure.Core;
+using TypeSpec.Rest;
+using TypeSpec.Http;
+
+namespace KeyVault;
+
+/**
+ * Operation status.
+ */
+@lroStatus
+union OperationStatus {
+  string,
+
+  /** The operation succeeded. */
+  @lroSucceeded
+  Success: "Success", // The correct response should be "Succeeded", but this has already shipped.
+
+  /** The operation is in progress. */
+  InProgress: "InProgress",
+
+  /** The operation failed. */
+  @lroFailed
+  Failed: "Failed",
+}
+
+/**
+ * The Security Domain operation status.
+ */
+model SecurityDomainOperationStatus {
+  /**
+   * Operation status.
+   */
+  status?: OperationStatus;
+
+  /**
+   * Details of the operation status.
+   */
+  @encodedName("application/json", "status_details")
+  statusDetails?: string;
+}
+
+/**
```

Comment:
These appear to be normal Azure Core errors. Can't we use those instead of redefining?

Reply:
The Key Vault error is a little different in its definition, but I do think it's functionally equivalent to the Azure Core error. If I'm not mistaken, KV's custom error has an additional `innerError` property.